Src/Support/Google.Apis.Auth/OAuth2/ImpersonatedCredential.cs

```diff
@@ -0,0 +1,264 @@
+                ct => RefreshOidcTokenAsync(tokenRefreshManager, options, ct), Clock, Logger);
+            return Task.FromResult(new OidcToken(tokenRefreshManager));
+        }
+
+        private async Task<bool> RefreshOidcTokenAsync(TokenRefreshManager caller, OidcTokenOptions options, CancellationToken cancellationToken)
+        {
+            var bodyJson = new {
+                delegates = this.Delegates,
+                audience = options.TargetAudience,
+                includeEmail = true
+            };
+            var body = NewtonsoftJsonSerializer.Instance.Serialize(bodyJson);
+            var content = new StringContent(body, Encoding.UTF8, "application/json");
+            
+            var oidcTokenUrl = String.Format(GoogleAuthConsts.IamIdTokenEndpoint, TargetPrincipal);
+            var response = await HttpClient.PostAsync(oidcTokenUrl, content, cancellationToken).ConfigureAwait(false);
+            caller.Token = await TokenResponse.FromHttpResponseAsync(response, Clock, Logger).ConfigureAwait(false);
+            return true;
+        }
+
+        /// <summary>
+        /// Signs the provided bytes using the private key associated with the impersonated service account.
+        /// </summary>
+        /// <param name="bytesToSign">The bytes to sign.</param>
+        /// <param name="cancellationToken">Cancellation token to cancel operation.</param>
+        /// <returns>The signed bytes.</returns>
+        /// <exception cref="SignBlobResponseException">When signing request fails.</exception>
+        public async Task<byte[]> SignBytes(byte[] bytesToSign, CancellationToken cancellationToken = default) {
+            var bodyJson = new {
+                delegates = this.Delegates,
+                payload = Convert.ToBase64String(bytesToSign)
+            };
+            var body = new StringContent(NewtonsoftJsonSerializer.Instance.Serialize(bodyJson), Encoding.UTF8, "application/json");
+            
+            var signBlobUrl = String.Format(GoogleAuthConsts.IamSignEndpoint, TargetPrincipal);
+            
+            var response = await HttpClient.PostAsync(signBlobUrl, body, cancellationToken).ConfigureAwait(false);
+            var content = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+            if (!response.IsSuccessStatusCode)
+            {
+                throw new SignBlobResponseException(content, response.StatusCode);
+            }
+            try
+            {
+                var signBlobResponse = NewtonsoftJsonSerializer.Instance.Deserialize<SignBlobResponse>(content);
+                return Convert.FromBase64String(signBlobResponse.SignedBlob);
+            }
+            catch (Newtonsoft.Json.JsonException)
+            {
+                throw new SignBlobResponseException("Server response does not contain a JSON object. Response content is: {content}", response.StatusCode);
+            }
+        }
+
+        /// <summary>
```

Comment:
Why do we need a whole new exception for this, and in particular *just* for impersonated credentials?

Reply:
We need an exception type for signblob response (TokenResponseException is the not the right one because signblob request is not a token request, also we would like the Error property to be a string type).  Which existing exception might be a good fit instead?

Reply:
I removed the new exception. Now using `response.EnsureSuccessStatusCode()`.